src/test/spec/unified_runner/operation.rs

```diff
@@ -2120,12 +2121,25 @@ impl TestOperation for Close {
     ) -> BoxFuture<'a, Result<Option<Entity>>> {
         async move {
             let mut entities = test_runner.entities.write().await;
-            let cursor = entities.get_mut(id).unwrap().as_mut_cursor();
-            let rx = cursor.make_kill_watcher().await;
-            *cursor = TestCursor::Closed;
-            drop(entities);
-            let _ = rx.await;
-            Ok(None)
+            let target_entity = entities.get_mut(id).unwrap();
+            match target_entity {
+                Entity::Client(client) => {
+                    client.client = ClientEntityState::Dropped;
+                    drop(entities);
```

Comment:
Hmm, I can't think of a reason either. This might have just been a mistake. If the tests pass without it, I'd say leave it out for cursors.

For clients, I think we do need it (or something similar) in order to drop all descendant collections / databases to ensure the client actually gets dropped.

Reply:
you raise a good point regarding the descendant databases and collections keeping a client alive. 

that said, re-reading this code, all `drop(entities)` does is drop the `RwLockWriteGuard` we acquired which gave exclusive write access to the entities. I'm not sure why that would have been needed for cursors , and I don't see why we would need it for clients, still but I'll try running a full patch with it removed for cursors to see if the test failures (or lack thereof) give any indication here.

in any case, I think we need to update the logic here to actually remove all descendant objects from the entity map so the client can be dropped, and I think I should update my spec PR to state that it is an error for a test to use any descendant object for an operation after its parent client is closed. 
there is already a sentence to this effect re using the client which I can expand on: 
> Test files SHOULD NOT
specify any operations for a client entity following a `close` operation on it, as
driver behavior when an operation is attempted on a closed client is not consistent.

Reply:
Ohh, I see now. I think it was dropped to avoid holding it across await point in case it caused a deadlock, since the guard constitutes exclusive access. Probably would have been good if I left a comment about that.

>in any case, I think we need to update the logic here to actually remove all descendant objects from the entity map so the client can be dropped

Sounds good :+1:

Reply:
Tests can run in parallel using the `runOnThread` operation introduced as part of the SDAM integration test port to the unified format. I'm not sure any test includes operations that would be problematic for this, but it's technically possible I think.